src/IMetrom.sol

```diff
@@ -37,36 +37,76 @@ struct RewardsCampaign {
 /// This can optionally contain a SHA256 hash of some JSON content stored on IPFS such
 /// that a CID can be constructed from it. `specification` can point to an IPFS JSON
 /// file with additional information/parameters on the campaign.
-struct PointsCampaign {
+struct PointsCampaignV1 {
     address owner;
     address pendingOwner;
     address pool;
     uint32 from;
     uint32 to;
-    bytes32 specification;
+    bytes32 specificationHash;
+    uint256 points;
+}
+
+/// @notice Represents a rewards based campaign in the contract's state, with its owner,
+/// running period, type, data, specification, root and data links, as well as rewards
+/// information. A particular note must be made for the `specification` and `data` fields.
+/// These can optionally contain a SHA256 hash of some JSON content stored on IPFS such that
+/// a CID can be constructed from them. `specification` can point to an IPFS JSON file with
+/// additional information/parameters on the campaign, while the `data` field must point
+/// to a JSON file containing the raw leaves from which the current campaign's Merkle
+/// tree and root was calculated.
+struct RewardsCampaignV2 {
+    address owner;
+    address pendingOwner;
+    uint32 from;
+    uint32 to;
+    uint32 kind;
+    bytes data;
+    bytes32 specificationHash;
+    bytes32 dataHash;
+    bytes32 root;
+    mapping(address token => Reward) reward;
+}
+
+/// @notice Represents a points based campaign in the contract's state, with its owner,
+/// running period, type, data, specification, root and data links, as well as rewards
```

Comment:
Minor change to the comment, for specificationHash and dataHash, also in the above structs.

Reply:
Addressed in 7d8eb3ae6abcac838c85ea9438846f6fc1156e8b